modules/core/shared/src/main/scala/eu/timepit/refined/numeric.scala

```diff
@@ -41,6 +41,9 @@ object numeric extends NumericInference {
   /** Predicate that checks if an integral value modulo `N` is `O`. */
   final case class Modulo[N, O](n: N, o: O)
 
+  /** Predicate that checks if a floating-point number value is not NaN. */
+  case class NotNaN()
```

Comment:
Other negated predicates use `Non` as prefix, like `NonEmpty`, `NonPositive`, `NonNegative`, or `NonDivisible`. I think we should do it here too and rename it to `NonNaN`. It [seems](https://en.wiktionary.org/wiki/non-) to be a common rule to prefix with non- to negate the meaning of a word.